### DISCUSSION TITLE
Pass command line parameters to flac encoder for replay gain tag

### DISCUSSION THREAD

**Discussion by Tnas12** [enzo1982/freac#418] [General Discussion]
Hello
As I use Fre:ac to encode my files to flac and from ape to flac...
I checked the "signal treatment" (english is not my native language ;-)
as I want to add replay gain to the flac files, and flac.exe can do it, 
I was wondering if is there a way to add command line parameters to flac encoder in fre:ac ?
-> the purpose is to add replay-gain :-)
-->> if possible to add the replay gain as an album but as the encoding is done in // I guess that it will not be possible 
or
it anyone as another way to do it within fre:ac that will be great

Thanks for the community help !

**Comment by enzo1982** [comment]
Hi,

fre:ac does not use flac.exe, but the FLAC codec library, so it is not possible to pass additional command line options to it. Also, the FLAC library does not have any ReplayGain code, that is implemented in the command line program.

So for now you will need to use a separate tool to generate Replay Gain tags.

I plan to add support for Replay Gain directly in fre:ac at some point, but cannot say yet when this will be available.

**Reply by seamusdemora** [reply]
@enzo1982 : FWIW, I think adding Replay Gain support would be a very useful addition to fre:ac. If you could entertain a question on that, how might you add this? Wouldn't it likely be as another library - to add the Replay Gain as a "final touch"?

**Reply by enzo1982** [reply]
The track gain can be calculated during conversion. This will probably be implemented as a signal processing filter that just measures the signal and internally computes the gain value. The filter component will then update the output file's tags with the Replay Gain information after the conversion is finished.

The last part is a bit tricky as filters do not usually alter file tags and the conversion engine is not yet prepared to handle this case.

For album gain, a component will need to keep track of all the gain values of files/tracks belonging to an album and then update them only after all tracks' gain values have been calculated.

**Reply by Tnas12** [reply]
Thanks Enzo for the clarification and help !

Great piece of software btw !

**Comment by kokosowy** [comment]
I also think replaygain is a must have feature. Since soundKonverter is no more being developed I am currently looking for an alternative and I am quite surprised probably most popular software for ripping/converting music for Linux is lacking this feature. So I think I will stick to fre:ac now, but I will need to find separate replaygain solution for a time being.

**Reply by kokosowy** [reply]
Did some research and now I will use [rsgain](https://wiki.hydrogenaud.io/index.php?title=Rsgain) for replaygian calculation - worked very well on my collection. Information for others interested.

**Comment by zjDrummond** [comment]
Just wanted to say that I'd love to be able to add the replaygain data with Freac. Currently I'm using MusicBrainz Picard on my Linux PC and Foobar2000 on my Windows PC to process files for replaygain as a second step when ripping CDs.
